examples/all-clusters-app/all-clusters-common/src/rvc-modes.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <rvc-modes.h>
+
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::RvcRunMode;
+using namespace chip::app::Clusters::RvcCleanMode;
+using chip::Protocols::InteractionModel::Status;
+
+CHIP_ERROR RvcRunModeInstance::AppInit()
+{
+    return CHIP_NO_ERROR;
+}
+
+void RvcRunModeInstance::HandleChangeToMode(uint8_t NewMode, ModeBase::Commands::ChangeToModeResponse::Type & response)
+{
+    uint8_t currentMode = GetCurrentMode();
+
+    // Our business logic states that we can only switch into the mapping state from the idle state.
+    if (NewMode == RvcRunMode::ModeMapping && currentMode != RvcRunMode::ModeIdle)
+    {
+        response.status = static_cast<uint8_t>(ModeBase::StatusCode::kGenericFailure);
+        response.statusText.SetValue(chip::CharSpan("Change to the mapping state is only allowed from idle", 53));
```

Comment:
```suggestion
        response.statusText.SetValue(chip::CharSpan("Change to the mapping mode is only allowed from idle", 53));
```

Also, number 53 should be replaced with something else (strlen or similar).

Reply:
Is there a reason why using `strlen` is preferred? Given how easy it is to count chars with modern IDEs, I think this makes it more readable. For me anyway. I can change it if it isn't.

Reply:
Just use `chip::CharSpan::fromCharString("string constant")` and then you don't have to specify a length at all and CharSpan will just do the right thing @hicklin

Reply:
Perfect! Thanks. It's done.